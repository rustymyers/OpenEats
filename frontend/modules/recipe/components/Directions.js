import React from 'react'

export default React.createClass({
  getInitialState: function() {
    return {
      data: this.props.data || []
    };
  },

  componentWillReceiveProps(nextProps) {
    this.setState({data: nextProps.data});
  },

  render: function() {
    var directions = this.state.data.map(function(direction) {
      return (
        <li className="direction" key={ direction.step }>
          { direction.title }
        </li>
      );
    });
    return (
      <ol className="directions" >
        { directions }
      </ol>
    );
  }
});
