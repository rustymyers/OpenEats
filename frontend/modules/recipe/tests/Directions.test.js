import React from 'react';
import Directions from '../components/Directions';
import renderer from 'react-test-renderer';

import data from './data';

test('Direction component test', () => {
  const component = renderer.create(
    <Directions data={ data.directions }/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
