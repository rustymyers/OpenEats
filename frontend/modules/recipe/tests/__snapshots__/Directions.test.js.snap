// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Direction component test 1`] = `
<ol
  className="directions"
>
  <li
    className="direction"
  >
    Brown the ground pork and ground sirlion in a medium pan. Add a teaspoon of sereano pepper while browning the meat. Season with kosher salt and pepper.
  </li>
  <li
    className="direction"
  >
    Chop the onion, bell pepper and one Serrano pepper and place them in a large pot.
  </li>
  <li
    className="direction"
  >
    Open up and drain both cans of kidney beans and add them to the large pot.
  </li>
  <li
    className="direction"
  >
    Open up both cans of stewed chili style tomatoes and add them to the pot.
  </li>
  <li
    className="direction"
  >
    Drain the grease away from the browned meat and add the meat to the pot.
  </li>
  <li
    className="direction"
  >
    Pour in the tomato juice over the meat mixture.
  </li>
  <li
    className="direction"
  >
    Add kosher salt, black pepper, two table spoons of chili powder, and two teaspoons of ground cumin. Stir well.
  </li>
  <li
    className="direction"
  >
    Cook slowly over medium low heat for an hour. If it starts to bubble turn down the heat. Taste during the cooking process to check the seasoning add more to taste.
  </li>
</ol>
`;
